interop-scoring/main.js

```diff
@@ -0,0 +1,506 @@
+      }
+    }
+  }
+  // category is an object with "name" and "labels" props.
+  for (const category of categories) {
+    console.log(`Scoring runs for ${category.name}`);
+    const testsSet = new Set();
+    // We aggregate all the tests with the labels defined in the category.
+    for (const label of category.labels) {
+      const labeledTestsSet = labeledTests.get(label);
+      if (!labeledTestsSet || !labeledTestsSet.size) {
+        throw new Error(`No tests labeled for ${label}`);
+      }
+      // Keep a unique set of tests associated with the category.
+      labeledTestsSet.forEach(test => testsSet.add(test));
+    }
+    const dateToScores = await scoreCategory(category, experimental, products,
+        alignedRuns, testsSet);
+    // Store the entire dateToScores for producing the unified CSV later.
+    dateToScoresMaps.set(category.name, dateToScores);
+  }
+
+  // TODO: Once the other score CSVs are no longer used, we can push
+  // some of this logic into scoreCategory and simplify things.
+  let unifiedCsv = 'date';
+  for (const product of products) {
+    const categoryLabels = categories.map(c => `${product}-${c.name}`);
+    unifiedCsv += `,${product}-version,${categoryLabels.join()}`;
+  }
+  // Add the interop category headers.
+  // An arbitrary interop version header is kept for ease of parsing on the interop dashboard.
+  unifiedCsv += `,interop-version,${categories.map(c => `interop-${c.name}`)}`;
+  unifiedCsv += '\n';
+
+  // We know that all dateToScoresMaps have the same dates (as they come from
+  // the same runs), so we can just iterate the keys from the first.
+  for (const date of dateToScoresMaps.get(categories[0].name).keys()) {
+    let csvLine = [date.substr(0, 10)];
+    // This is essentially an inversion loop; we have the data mapped by
+    // individual categories, but we need it mapped by product.
+    for (let browserIdx = 0; browserIdx < products.length; browserIdx++) {
+      let version;
+      const productScores = [];
+      for (const category of categories) {
+        const {versions, scores} = dateToScoresMaps.get(category.name).get(date);
+        const score = scores[browserIdx];
+        productScores.push(score);
+        // The versions should all be the same, so we just grab the latest one.
+        version = versions[browserIdx];
+      }
+      csvLine.push(version);
+      csvLine = csvLine.concat(productScores);
+    }
+    // Add the interop scores for each category.
+    csvLine.push('-');
+    for (const category of categories) {
+      const scoreInfo = dateToScoresMaps.get(category.name).get(date);
+      const categoryInteropScore = scoreInfo.scores[products.length];
+      csvLine.push(categoryInteropScore);
+    }
+    unifiedCsv += `${csvLine.join()}\n`;
+  }
+
+  // Use "-v2" suffix to differentiate from the old csv formats.
+  const csvFilename = experimental ?
+      `interop-${year}-experimental-v2.csv` : `interop-${year}-stable-v2.csv`;
```

Comment:
`-v2` suffix is added to differentiate between a csv created with the older scripts.